multimodal/tests/unittests/test_distiller.py

```diff
@@ -86,3 +86,32 @@ def test_distillation():
         save_path=student_save_path,
     )
     verify_predictor_save_load(predictor, dataset.test_df)
+
+    # test for distillation with different model structures
+    different_hyperparameters = {
+        "optimization.max_epochs": 1,
+        "model.names": ["hf_text", "timm_image", "fusion_mlp"],
+        "model.hf_text.checkpoint_name": "google/electra-small-discriminator",
+        "model.timm_image.checkpoint_name": "swin_tiny_patch4_window7_224",
+        "env.num_workers": 0,
+        "env.num_workers_evaluation": 0,
+    }
+
+    predictor = MultiModalPredictor(
+        label=dataset.label_columns[0],
+        problem_type=dataset.problem_type,
+        eval_metric=dataset.metric,
+    )
+
+    student_save_path = os.path.join(get_home_dir(), "petfinder", "student")
+    if os.path.exists(student_save_path):
+        shutil.rmtree(student_save_path)
+
+    predictor = predictor.fit(
+        train_data=dataset.train_df,
+        teacher_predictor=teacher_predictor,
+        hyperparameters=different_hyperparameters,
+        time_limit=30,
+        save_path=student_save_path,
+    )
+    verify_predictor_save_load(predictor, dataset.test_df)
```

Comment:
Why do we need add this new test? This code block have significant overlaps with the previous.

Reply:
I added this to test if the adaptor works fine when teacher and student have different output feature shape.

Reply:
We can replace the old test with the new one. No need to keep both.

Reply:
Done